src/api/Nhs.Appointments.Persistance/SiteStore.cs

```diff
@@ -19,16 +19,34 @@ public Task<IEnumerable<SiteWithDistance>> GetSitesByArea(double longitude, doub
         return cosmosStore.RunSqlQueryAsync<SiteWithDistance>(query);
     }
     
-    public async Task<Site> GetSiteById(string siteId)
+    public Task<Site> GetSiteById(string siteId)
+    {
+        return GetOrDefault(siteId);
+    }
+    
+    public async Task<bool> UpdateSiteAttributes(string siteId, IEnumerable<AttributeValue> attributeValues)
+    {
+        var originalDocument = await GetOrDefault(siteId);
+        if (originalDocument == null)
+        {
+            return false;
+        }
```

Comment:
Just musing aloud here, but I see you've ran into exactly the same decision I had to make recently with the RemoveUserFunction. I considered a `Task<bool>` as well but because there were multiple reasons it could fail, I reasoned that a bool wasn't sufficient. I know Vince isn't entirely happy with the `OperationResult` concept and tbh neither am I, and now we're introducing some inconsistency so perhaps there's some future work here for a tech debt afternoon? 

(If we can agree on the best way of handling this situation, that is! 😄)